### PR TITLE
rename confusing functions

### PR DESCRIPTION
**Pull request by banzay**
Rename confusing function extract_user to extract_user_key
#33

### CODE REVIEW

File: lib/gardien.ex [hunk 1]
```diff
@@ -39,25 +40,41 @@ defmodule Gardien do
   defp do_extract_options(%{resource: nil}), do: raise(MisssingOptionsError, :resource)
   defp do_extract_options(%{policy: nil}), do: raise(MisssingOptionsError, :policy)
   defp do_extract_options(options) do
-    resource       = extract_resource(options)
-    policy         = extract_policy(options)
-    unauth_handler = extract_unauth_handler(options)
-    user           = extract_user(options)
+    %{
+       resource: extract_resource_option(options),
+       policy: extract_policy_option(options),
```
Comment (rpelyush) [inline lib/gardien.ex]: ```extract_policy``` was fine, because here we do extract actual policy.

File: lib/gardien.ex [hunk 2]
```diff
@@ -39,25 +40,41 @@ defmodule Gardien do
...
+      handler when is_function(handler) ->
+        handler
+      {_mod, _func} = handler ->
+        handler
+      _ ->
+        Application.get_env(:gardien, :unauth_handler, &Gardien.default_unauth_handler/1)
+    end
+  end
 
+  # Extract data from conn. Executed at runtime.
```
Comment (rpelyush) [inline lib/gardien.ex]: This comment is unnecessary.

File: lib/gardien.ex [hunk 3]
```diff
@@ -29,6 +29,7 @@ defmodule Gardien do
     authorize(conn, user, resource, policy, action, unauth_handler)
   end
 
+  # Extract data from options. Executed at compile-time.
```
Comment (rpelyush) [inline lib/gardien.ex]: Remove this comments please. Code should be self-explanatory.

**Comment by banzay** [comment]
done

**Comment by banzay** [comment]
done

File: lib/gardien.ex [hunk 4]
```diff
@@ -39,33 +39,14 @@ defmodule Gardien do
...
-
-  defp extract_user(%{current_user: user} = _options) when is_nil(user) do
-    Application.get_env(:gardien, :current_user, :current_user)
-  end
-  defp extract_user(%{current_user: user} = _options), do: user
-
-  defp extract_user(conn, %{current_user: user} = _options) do
-    Map.get(conn.assigns, user)
+    %{
+       resource: Map.get(options, :resource),
```
Comment (rpelyush) [inline lib/gardien.ex]: Why did you remove ```extract_resource(options)```? It was my point to wrap each step in a separate function. This way logic related to extracting resource option is concentrated in dedicated function.

Comment (rpelyush) [inline lib/gardien.ex]: please add ```extract_resource_option``` function.

Comment (banzay) [inline lib/gardien.ex]: done

**Comment by rpelyush** [comment]
👍
